### PR TITLE
context/core/netty: Add @CheckReturnValue to Context

### PR DESCRIPTION
**Pull request by dapengzhang0**
This change is [Reviewable](https://reviewable.io/reviews/grpc/grpc-java/4430)

### CODE REVIEW

**Comment by carl-mastrangelo** [comment]
This needs to be check with an internal presubmit first.

**Comment by dapengzhang0** [comment]
> This needs to be check with an internal presubmit first.
Yes, this change is scary.

**Comment by dapengzhang0** [comment]
The internal presubmit (CL/195275931) passed.

**Comment by carl-mastrangelo** [comment]
One other concern: this package has traditionally not had any dependencies (including DoNotMock).   @adriancole I recall you being involved in the discussion around deps of Context.  Would taking a compile time dependency on ErrorProne annotations be feasible?

**Comment by dapengzhang0** [comment]
> One other concern: this package has traditionally not had any dependencies (including DoNotMock). @adriancole I recall you being involved in the discussion around deps of Context. Would taking a compile time dependency on ErrorProne annotations be feasible?

One of Context users, Zipkin instrumentation libs, has no deps, not even annotation jars. So this PR is deliberately not adding any dependencies. So `com.google.errorprone.annotations.CheckReturnValue` can not be used in `io.grpc.Context`. Instead, it uses `io.grpc.Context.CheckReturnValue`.  It is up to the compiler, whether or not to apply the errorprone plug-in when compiling (in static analysis phase). Internally, the JavaBuilder will apply errorprone by default for all projects.

File: context/src/main/java/io/grpc/Context.java [hunk 1]
```diff
@@ -1059,4 +1064,8 @@ private static void validateGeneration(int generation) {
           new Exception());
     }
   }
+
```
Comment (carl-mastrangelo) [inline context/src/main/java/io/grpc/Context.java]: This probably deserves a comment why it isn't the error prone version.

**Comment by dapengzhang0** [comment]
Thanks for the review @carl-mastrangelo .

**Pull request merged**